designs/conditional-controllers.md

```diff
@@ -0,0 +1,174 @@
+Informer Removal / Controller Lifecycle Management
+==========================
+
+## Summary
+
+Enable fine-grained control over the lifecycle of a controller, including the
+ability to start/stop/restart controllers and their caches by exposing a way to
+remove individual informers from the cache and working around restrictions that
+currently prevent controllers from starting multiple times.
```

Comment:
Is supporting this in multi-cluster scenarios a goal?

Reply:
My understanding is that there's isn't anything explicitly preventing a single-cluster solution from working for multiple clusters out of the box. It's sounds like you have specific concerns around this approach not working multi-cluster.

It's just that because controller-runtime historically has not attempted to support multi-cluster, attempting to use any solution that comes from this effort in a multi-cluster manner would be harder and undocumented.

I think the short answer is no, we aren't looking to fully flesh out a multi-cluster solution here but @DirectXMan12 can chime in if I'm misspeaking

Reply:
I guess if this is a generic question, it doesn't really matter what cluster a given controller talks to (But I can't mark this convo as resolved)